packages/core/src/sync-store/postgres/store.ts

```diff
@@ -669,6 +677,225 @@ export class PostgresSyncStore implements SyncStore {
     return hasBlock !== undefined;
   };
 
+  insertTraceFilterInterval = async ({
+    chainId,
+    traceFilter,
+    block: rpcBlock,
+    transactions: rpcTransactions,
+    transactionReceipts: rpcTransactionReceipts,
+    traces: rpcTraces,
+    interval,
+  }: {
+    chainId: number;
+    traceFilter: CallTraceFilterCriteria;
+    block: RpcBlock;
+    transactions: RpcTransaction[];
+    transactionReceipts: RpcTransactionReceipt[];
+    traces: SyncCallTrace[];
+    interval: { startBlock: bigint; endBlock: bigint };
+  }) => {
+    return this.db.wrap({ method: "insertTraceFilterInterval" }, async () => {
+      await this.db.transaction().execute(async (tx) => {
+        await tx
+          .insertInto("blocks")
+          .values({
+            ...rpcToPostgresBlock(rpcBlock),
+            chainId,
+            checkpoint: this.createBlockCheckpoint(rpcBlock, chainId),
+          })
+          .onConflict((oc) => oc.column("hash").doNothing())
+          .execute();
+
+        if (rpcTransactions.length > 0) {
+          const transactions = rpcTransactions.map((transaction) => ({
+            ...rpcToPostgresTransaction(transaction),
+            chainId,
+          }));
+          await tx
+            .insertInto("transactions")
+            .values(transactions)
+            .onConflict((oc) => oc.column("hash").doNothing())
+            .execute();
+        }
+
+        if (rpcTransactionReceipts.length > 0) {
+          const transactionReceipts = rpcTransactionReceipts.map(
+            (rpcTransactionReceipt) => ({
+              ...rpcToPostgresTransactionReceipt(rpcTransactionReceipt),
+              chainId,
+            }),
+          );
+          await tx
+            .insertInto("transactionReceipts")
+            .values(transactionReceipts)
+            .onConflict((oc) => oc.column("transactionHash").doNothing())
+            .execute();
+        }
+
+        // Delete existing traces with the same `transactionHash`. Then, calculate "callTraces.checkpoint"
+        // based on the ordering of "callTraces.traceAddress" and add all traces to "callTraces" table.
+        const traceByTransactionHash: {
+          [transacionHash: Hex]: SyncCallTrace[];
+        } = {};
+        for (const trace of rpcTraces) {
+          if (traceByTransactionHash[trace.transactionHash] === undefined) {
+            traceByTransactionHash[trace.transactionHash] = [];
+          }
+          traceByTransactionHash[trace.transactionHash].push(trace);
+        }
+
+        for (const transactionHash of Object.keys(traceByTransactionHash)) {
+          const traces = await tx
```

Comment:
I'm not sure, but it might be worth batching this using `.where("transactionHash", "in", transactionHashes)`, doing the reordering in-memory, and then inserting all of the merged traces in one query at the end. It's plausible that N could be >100 here.